source/docs/software/vision-processing/introduction/game-piece-detection.rst

```diff
@@ -0,0 +1,185 @@
+.. include:: <isonum.txt>
+
+# Identifying Game Pieces
+
+A common task in FRC\ |reg| is locating and navigating to game pieces on the field. This document describes approaches for detecting game pieces using vision processing and converting that information into actionable data for your robot.
+
+## Detection Methods: Traditional Vision vs Machine Learning
+
+There are two primary approaches to detecting game pieces: traditional computer vision and machine learning (AI) models. Choosing the right approach depends on your specific use case.
+
+### When to Use Traditional Vision Methods
+
+Traditional vision methods work best when:
+
+- The game piece has a **distinctive, consistent color** that stands out from the background
+- The object looks **similar from all angles** (e.g., a sphere or simple geometric shape)
+- **Lighting conditions are controlled** (e.g., using ring lights or consistent field lighting)
+- You want **fast, predictable performance** with lower computational requirements
+
+Examples: Detecting brightly colored balls, simple geometric shapes with consistent colors
+
+### When to Use Machine Learning
+
+Machine learning models are better suited when:
+
+- The game piece has **variable appearance** depending on viewing angle
+- The object has **complex geometry** or texture that's hard to describe with simple color rules
+- There is **visual variability** in the objects themselves (wear, damage, manufacturing differences)
+- Background conditions vary significantly
+
+Examples: Detecting cargo with logos, irregularly shaped objects, pieces that look very different when viewed from different angles
+
+.. note:: Most common FRC vision processing vendors (PhotonVision, Limelight, etc.) offer support for both traditional vision pipelines and machine learning models. The choice of method should be based on your specific game piece characteristics, not the limitations of your hardware.
```

Comment:
Is ML without some type of accelerator feasible? I thought it was slow enough to not be very usable. I agree with the statement that you shouldn't choose ML because you have a cool coprocessor, but I think you might want to choose a different coprocessor if you need to use ML, so it's probably worth a mention about choosing a coprocessor for ML.